### ISSUE TITLE
Unable to sync notes content in drive

### ISSUE THREAD

**Issue by bhupesh-sf** [penge/my-notes#302]
I just noticed a behavior where notes content is not being synced. If I rename a note, delete a note that's gets synced but if I update the content it doesn't.

I had a new laptop and there also I used my notes on chrome, it synced all the notes but the content was 2 months old whereas on my main laptop in chrome it is showing latest content. I tried to make changes to the notes but nothing was reflecting in google drive (I went to drive to check the content as well).

Its creating problem while syncing notes among multiple devices. So please fix it at priority.

**Comment by penge** [comment]
Do you sync manually (click on Sync now button) or automatically (Auto Sync)? None of the two is working?
Do you have notes with same name in Google Drive? Or maybe with (1) or (2) or similar suffixes?

**Comment by bhupesh-sf** [comment]
I did it manually as well but its not working and there is no new note with (1) or (2) suffixes.

**Comment by penge** [comment]
When you click on Sync now button, will it spin? And stop spinning after a short moment?
Is Synced information under the info icon (bottom right) having old date, or recent? Is this date changing when you sync?

**Comment by bhupesh-sf** [comment]
It works all as expected. Sync button will spin, then stop after some time and sync information like date and time gets updated as well.

I told you the problem is content is not being synced but if you rename or delete or create a note then it gets perfectly. The problem is with content syncing.

**Comment by penge** [comment]
OK. I'll look into this.

**Comment by penge** [comment]
@bhupesh-sf 
Can you run this in Chrome console in My Notes tab:
```
chrome.storage.local.getBytesInUse("notes", (result) => console.log(result))
```
and paste here the result?

**Comment by bhupesh-sf** [comment]
234289

**Comment by penge** [comment]
That looks fine. Limit is 5000000 (5MB).

What do you see on the second laptop? Can you check that? Do you have some notes with images?

Do you see following error in either of laptops in the console?

```
Unchecked runtime.lastError: QUOTA_BYTES quota exceeded
```

It can look like this:

<img width="1792" alt="QUOTA_BYTES quota exceeded" src="https://user-images.githubusercontent.com/907255/130813334-9b6b34d2-4cdd-4bc2-b09a-1e5875844cf5.png">

**Comment by penge** [comment]
Sounds cool, if it occurs again, let me know. I will meanwhile try to replicate that problem.

**Comment by bhupesh-sf** [comment]
I experienced the same again here.

On my ubuntu machine, I made changes to a note. Then I saw drive sync was disabled which I never did (not sure how it happened). When I enabled it, it started syncing that note. After that what I see is note content is duplicated (in the same note). On top, it contains the note with all the changes which I did, and just below what I see is the old copy before changes (seems this is what the last sync was). Then I opened my MAC and checked there. I synced everything but it still shows me the old copy without any new changes. 

I also created a new note on the ubuntu machine which is perfectly synced on my MAC.

So in a nutshell problem seems to be that when you make some changes to an existing note it doesn't sync those changes to the drive rather it appends the same old content of the drive below the changes and hence creating copies of the note. New changes remain on local and are never synced on the drive.

**Comment by bhupesh-sf** [comment]
@penge Please see this issue is important as I am going to transition to MAC from ubuntu soon and I need this content synced there.

**Comment by penge** [comment]
It seems that after installing Google Chrome updates, it may **disable** (in other words require to be enabled again) a permission that is needed for Google Drive Sync. This permission is called **"identity"** and it's optional.

I am not sure if there's something I can do about it other than making **"identity"** permission **required**, which could solve it. My Notes have nowhere in code something that could remove this permission accidentally.

The reason for having an old copy of note **added to the end** of note, is a side effect caused by disabled/enabled Google Drive Sync in case **"last modified time"** of local note and note in Google Drive, are different. To make sure one note does not override the other and nothing is lost, contents are merged. If content was _NOT_ modified since the Google Drive Sync was disabled, old content should _NOT_ be copied of top the other when Google Drive Sync is enabled again.

I'd like to also check why you had an old copy of note on your Mac. The best really could be, if Google Drive Sync doesn't disable itself just like that. Maybe what I could do, is to show a popup: Google Drive Sync was disabled since your last visit. Possible reason: Chrome updates. Would you like to enable Google Drive Sync again? And after resolving this popup with either Yes or No, you would continue editing notes. This way, we could somehow avoid any troubles where you edit notes, but Sync was disabled without notice, and updates were not synced. Although looking at Sync Now button can tell the same, it can be easy to overlook when you expect it to be enabled.

**Comment by bhupesh-sf** [comment]
I can understand your analysis and maybe we can implement this solution of notifying user regarding the google drive sync disabled.
